lib/rbs/json_schema/generator.rb

```diff
@@ -0,0 +1,453 @@
+            array.each do |definition|
+              # Parse schema & append extracted type to union's types
+              if !(temp = translate_type(uri, definition)).nil?
+                type.types << temp
+              end
+            end
+          else
+            raise ValidationError.new(message: "Invalid JSON Schema: #{SCHEMA_DEFINITIONS[6]}: #{schema[SCHEMA_DEFINITIONS[6]]}")
+          end
+        # SCHEMA_DEFINITIONS[5] => allOf => An intersection of types
+        when schema.key?(SCHEMA_DEFINITIONS[5])
+          # Assign intersection type
+          type = fetch_type("intersection")
+          if (array = schema[SCHEMA_DEFINITIONS[5]]).is_a?(Array)
+            # Iterate over each schema & append extracted type to interseciton's types
+            array.each do |definition|
+              # Parse schema & append extracted type to union's types
+              if !(temp = translate_type(uri, definition)).nil?
+                type.types << temp
+              end
+            end
+          else
+            raise ValidationError.new(message: "Invalid JSON Schema: #{SCHEMA_DEFINITIONS[5]}: #{schema[SCHEMA_DEFINITIONS[5]]}")
+          end
+        # SCHEMA_DEFINITIONS[2] => type => any type permitted by JSON schema
+        when schema.key?(SCHEMA_DEFINITIONS[2])
+          case schema[SCHEMA_DEFINITIONS[2]]
+          when "integer"
+            # If type = "integer", assign a class instance of integer for RBS
+            type = fetch_type("class_instance", name: "Integer", absolute: true)
+          when "number"
+            # If type = "number", assign a class instance of numeric for RBS
+            type = fetch_type("class_instance", name: "Numeric", absolute: true)
+          when "string"
+            # If type = "string", assign a class instance of string for RBS
+            type = fetch_type("class_instance", name: "String", absolute: true)
+          when "boolean"
+            # If type = "boolean", assign a base type of bool for RBS
+            type = fetch_type("boolean")
+          when "null"
+            # If type = "null", assign a base type of nil for RBS
+            type = fetch_type("nil")
+          when "object", "array"
+            # nop
+          else
+            raise ValidationError.new(message: "Invalid JSON Schema: #{SCHEMA_DEFINITIONS[2]}: #{schema[SCHEMA_DEFINITIONS[2]]}")
+          end
+        # SCHEMA_DEFINITIONS[8] => $ref => An alias type
+        when schema.key?(SCHEMA_DEFINITIONS[8])
+          begin
+            # Parse URI of `$ref`
+            ref_uri = URI.parse(schema[SCHEMA_DEFINITIONS[8]])
+          rescue URI::InvalidURIError => _
+            raise ValidationError.new(message: "Invalid URI encountered in: #{SCHEMA_DEFINITIONS[8]} = #{schema[SCHEMA_DEFINITIONS[8]]}")
+          end
+
+          resolved_uri = resolve_uri(uri, ref_uri) # Resolve `$ref` URI with respect to current URI
+          if resolved_uri.path == uri.path
```

Comment:
I don't think we should skip generating definitions for other files, but save the declarations included each JSON files in Hash.

```
attr_reader path_decls: Hash[String,AST::Declarations::Module]
```

Then we can group the declarations of each JSON file in the output.

```ruby
# generate_rbs
@path_decls[uri.path].members << decl if !decl.type.nil?
```